tests/sentry/eventstream/kafka/test_consumer.py

```diff
@@ -0,0 +1,632 @@
+        assert message.offset() == expected_message.offset()
+
+        # We should not be able to continue reading into the topic.
+        # TODO: Can we assert that the position is paused?
+        assert consumer.poll(1) is None
+
+
+@requires_kafka_client
+def test_consumer_start_from_committed_offset():
+    consumer_group = 'consumer-{}'.format(uuid.uuid1().hex)
+    synchronize_commit_group = 'consumer-{}'.format(uuid.uuid1().hex)
+
+    messages_delivered = defaultdict(list)
+
+    def record_message_delivered(error, message):
+        assert error is None
+        messages_delivered[message.topic()].append(message)
+
+    producer = Producer({
+        'bootstrap.servers': os.environ['SENTRY_KAFKA_HOSTS'],
+        'on_delivery': record_message_delivered,
+    })
+
+    with create_topic() as topic, create_topic() as commit_log_topic:
+
+        # Produce some messages into the topic.
+        for i in range(3):
+            producer.produce(topic, '{}'.format(i).encode('utf8'))
+
+        assert producer.flush(5) == 0, 'producer did not successfully flush queue'
+
+        Consumer({
+            'bootstrap.servers': os.environ['SENTRY_KAFKA_HOSTS'],
+            'group.id': consumer_group,
+        }).commit(
+            message=messages_delivered[topic][0],
+            asynchronous=False,
+        )
+
+        # Create the synchronized consumer.
+        consumer = SynchronizedConsumer(
+            bootstrap_servers=os.environ['SENTRY_KAFKA_HOSTS'],
+            consumer_group=consumer_group,
+            commit_log_topic=commit_log_topic,
+            synchronize_commit_group=synchronize_commit_group,
+            initial_offset_reset='earliest',
+        )
+
+        assignments_received = []
+
+        def on_assign(c, assignment):
+            assert c is consumer
+            assignments_received.append(assignment)
+
+        consumer.subscribe([topic], on_assign=on_assign)
+
+        # Wait until we have received our assignments.
+        for i in xrange(10):  # this takes a while
+            assert consumer.poll(1) is None
+            if assignments_received:
+                break
+
+        assert len(assignments_received) == 1, 'expected to receive partition assignment'
+        assert set((i.topic, i.partition) for i in assignments_received[0]) == set([(topic, 0)])
+
+        # TODO: Make sure that all partitions are paused on assignment.
+
+        # Move the committed offset forward for our synchronizing group.
+        message = messages_delivered[topic][0]
+        producer.produce(
+            commit_log_topic,
+            key='{}:{}:{}'.format(
+                message.topic(),
+                message.partition(),
+                synchronize_commit_group,
+            ).encode('utf8'),
+            value='{}'.format(
+                message.offset() + 1,
+            ).encode('utf8'),
+        )
+
+        # Make sure that there are no messages ready to consume.
+        assert consumer.poll(1) is None
```

Comment:
Hmm, now I'm also not sure about this either. shouldn't we expect the consumer to get message 0 here? as message 0's offset has just been sent to the synchronize_commit_group?

Reply:
oh, I guess we havent flushed to `synchronize_commit_group` at this stage in the test. still I don't get why the consumer never gets message 0